ldms/src/sampler/zfs_common/zfs_common.h

```diff
@@ -0,0 +1,26 @@
+
+/* -*- c-basic-offser: 8 -*- */
+/* Copyright 2022 Lawrence Livermore National Security, LLC
+ * See the top-level COPYING file for details.
+ *
+ * SPDX-License-Identifier: (GPL-2.0 OR BSD-3-Clause)
+ */
+
+#ifndef ZFS_COMMON
+#define ZFS_COMMON
+
+#ifndef ARRAY_LEN
+#define ARRAY_LEN(a) (sizeof(a) / sizeof(*a))
+#endif
+
+#define DEFAULT_ARRAY_LEN 32
+
+#endif
+
+/* prototypes */
+
+char * escape_string(char *s);
+
+int get_vdevs_count(zpool_handle_t *zhp, void *data);
```

Comment:
This header needs proper #include statements to define zpool_handle_t and nvlist_t.